pkg/attestation/server.go

```diff
@@ -0,0 +1,239 @@
+	}
+
+	// Get AK public from init's AK params
+	_, akParams, err := s.ParseInit(init)
+	if err != nil {
+		return VerificationResult{}, err
+	}
+	// Convert AK public to crypto.PublicKey
+	akPub, err := akPublicFromParams(akParams)
+	if err != nil {
+		return VerificationResult{}, err
+	}
+
+	// Parse and verify the PCR quote
+	var pq struct {
+		Quote struct {
+			Version   string `json:"version"`
+			Quote     []byte `json:"quote"`
+			Signature []byte `json:"signature"`
+		} `json:"quote"`
+		PCRs map[int][]byte `json:"pcrs"`
+	}
+	if err := json.Unmarshal(proof.PCRQuote, &pq); err != nil {
+		return VerificationResult{}, fmt.Errorf("unmarshaling PCR quote: %w", err)
+	}
+
+	// Verify the PCR quote signature and extract verified PCR values using tpm-helpers
+	verifiedPCRs, err := tpm.VerifyPCRQuote(proof.PCRQuote, akPub)
+	if err != nil {
+		return VerificationResult{}, fmt.Errorf("PCR quote verification failed: %w", err)
+	}
+
+	return VerificationResult{AKPublic: akPub, PCRs: verifiedPCRs}, nil
+}
+
+func (s *RemoteAttestationServer) IssuePassphrase(ctx context.Context, init *AttestationInit, proof *AttestationProof, expectedSecret []byte) ([]byte, error) {
+	vr, err := s.VerifyProof(init, proof, expectedSecret)
+	if err != nil {
+		return nil, err
+	}
+	// Derive TPM hash from EK again for input to Attestator
+	ek, _, err := s.ParseInit(init)
+	if err != nil {
+		return nil, err
+	}
+	tpmHash, err := ComputeTPMHashFromEK(ek)
+	if err != nil {
+		return nil, err
+	}
+	// Encode EK to PEM for the attestator
+	ekPEM, err := EncodeEKToPEM(ek)
+	if err != nil {
+		return nil, fmt.Errorf("encoding EK to PEM: %w", err)
+	}
+
+	// Build request for Attestator
+	req := AttestationRequest{
+		TPMHash:            tpmHash,
+		PCRs:               vr.PCRs,
+		EKPEM:              ekPEM,
+		DeferPCREnrollment: init.DeferPCREnrollment,
+	}
+	return s.attestator.IssuePassphrase(ctx, req)
+}
+
+// Helpers
+func equalBytes(a, b []byte) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	for i := range a {
+		if a[i] != b[i] {
+			return false
+		}
+	}
+	return true
+}
+
+func ComputeTPMHashFromEK(ek *attest.EK) (string, error) {
+	spki, err := x509.MarshalPKIXPublicKey(ek.Public)
+	if err != nil {
+		return "", fmt.Errorf("marshal EK: %w", err)
+	}
+	sum := sha256.Sum256(spki)
+	return fmt.Sprintf("%x", sum[:]), nil
+}
+
+func akPublicFromParams(params *attest.AttestationParameters) (crypto.PublicKey, error) {
+	pub, err := tpm2.Unmarshal[tpm2.TPMTPublic](params.Public)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal TPM public: %w", err)
+	}
+	switch pub.Type {
+	case tpm2.TPMAlgRSA:
+		rsaParms, err := pub.Parameters.RSADetail()
+		if err != nil {
+			return nil, fmt.Errorf("rsa params: %w", err)
+		}
+		rsaUnique, err := pub.Unique.RSA()
+		if err != nil {
+			return nil, fmt.Errorf("rsa unique: %w", err)
+		}
+		n := new(big.Int).SetBytes(rsaUnique.Buffer)
+		e := int(rsaParms.Exponent)
+		if e == 0 {
+			e = 65537
```

Comment:
magic number, at least a small comment here to say that this is the default rsa exponent if not configured?